From ed4fe56ef8724adeb20dcdc7d09503d002ba7bd9 Mon Sep 17 00:00:00 2001
From: Peter Robinson <pbrobinson@gmail.com>
Date: Sun, 9 Sep 2018 18:24:06 +0100
Subject: [PATCH] Use IdleNotificationDeadline as deprecated IdleNotification
 has been removed

In v8 API 6.5 the long deprecated IdleNotification has been removed so we
need to use IdleNotificationDeadline as nodejs 10.x uses newer v8.

Signed-off-by: Peter Robinson <pbrobinson@gmail.com>
Signed-off-by: Noel Eck <noel.eck@intel.com>
---
 src/common_top.i | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/common_top.i b/src/common_top.i
index c9edb5851..ae2b904e6 100644
--- a/src/common_top.i
+++ b/src/common_top.i
@@ -28,9 +28,9 @@ void cleanUp()
      * See https://codereview.chromium.org/412163003 for this API change
      */
 #if (SWIG_V8_VERSION < 0x032838)
-    while (!v8::V8::IdleNotification())
+    while (!v8::V8::IdleNotificationDeadline())
 #else
-    while (!v8::Isolate::GetCurrent()->IdleNotification(1000))
+    while (!v8::Isolate::GetCurrent()->IdleNotificationDeadline(1000))
 #endif
     {;}
 }
